src/librustc/ty/layout.rs

```diff
@@ -649,23 +649,21 @@ impl Scalar {
         }
     }
 
-    /// Returns a range suitable to be passed to LLVM for range metadata.
-    pub fn range_metadata<C: HasDataLayout>(&self, cx: C) -> Option<Range<u128>> {
+    /// Returns the valid range as a `x..y` range.
+    ///
+    /// If `x` and `y` are equal, the range is full, not empty.
+    pub fn range_metadata_exclusive<C: HasDataLayout>(&self, cx: C) -> Range<u128> {
```

Comment:
Please name this `valid_range_exclusive`.

Reply:
Renamed and squashed everything.